spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -630,14 +630,22 @@ def setup_editor(self,
                      occurrence_timeout=1500,
                      show_class_func_dropdown=False,
                      indent_guides=False,
-                     scroll_past_end=False):
+                     scroll_past_end=False, 
+                     debug_panel=True,
+                     folding=True):
 
         self.set_close_parentheses_enabled(close_parentheses)
         self.set_close_quotes_enabled(close_quotes)
         self.set_add_colons_enabled(add_colons)
         self.set_auto_unindent_enabled(auto_unindent)
         self.set_indent_chars(indent_chars)
 
+        # Hide debug panel depending on the language
+        self.set_debug_panel(debug_panel, language)
+
+        # Hide folding panel depending on the language
+        self.set_folding_panel(folding, language)

```

Comment:
@juanis2112 here please only pass as parameter `folding` and update the comment to reflect that. Although, should we do a validation for the folding panel with a `language` parameter as with the debug panel @ccordoba12 ?

Reply:
No, I don't think that's necessary because we plan to add more folding in beta3.